core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -325,7 +323,7 @@ public Block copyPositions(int[] positions, int offset, int length)
         checkArrayRange(positions, offset, length);
 
         IntArrayList positionsToCopy = new IntArrayList();
-        Map<Integer, Integer> oldIndexToNewIndex = new HashMap<>();
+        Int2IntOpenHashMap oldIndexToNewIndex = new Int2IntOpenHashMap();
```

Comment:
If you initialize the hash map to ```length``` initialize size, you may be able to avoid several expensive hash table resizing operations.